### PR TITLE
visibility dropdown private public yale community only

### PR DESCRIPTION
**Pull request by MaggieZhaoYale**
https://app.zenhub.com/workspaces/yul-dc-5e50083561915b11fc9e5850/issues/yalelibrary/yul-dc/607
Added: 

- 3 levels visibilities:  'Private', 'Public', 'Yale Community Only' in visibility dropdown (phrase1)

- validates :visibility in model of parent_object

Screen Shot 2020-09-23 at 2 04 48 PM

### CODE REVIEW

File: app/models/parent_object.rb [hunk 1]
```diff
@@ -9,6 +9,8 @@ class ParentObject < ApplicationRecord # rubocop:disable Metrics/ClassLength
   has_many :child_objects, primary_key: 'oid', foreign_key: 'parent_object_oid', dependent: :destroy
   belongs_to :authoritative_metadata_source, class_name: "MetadataSource"
   attr_accessor :metadata_update
+  validates :visibility, inclusion: { in: ['Private', 'Public', 'Yale Community Only'],
+                                      message: "%{value} is not a valid value" }
```
Comment (orangewolf) [inline app/models/parent_object.rb]: please add a class method called visibilities like so:

```
def self.visibilities
  ['Private', 'Public', 'Yale Community Only']
end
```
then use that class method instead of the hard coded string here and in the view. That way, if we ever have to add one, we can easily

File: app/views/parent_objects/_form.html.erb [hunk 2]
```diff
@@ -63,7 +63,7 @@
 
   <div class="field">
     <%= form.label :visibility %>
-    <%= form.text_field :visibility %>
+    <%= form. select(:visibility,[['Private','Private'],['Public','Public'],['Yale Community Only','Yale Community Only']] ) %>
```
Comment (orangewolf) [inline app/views/parent_objects/_form.html.erb]: I think you can declair these once `['Private', 'Public', 'Yale Community Only']` and Rails will automatically do the doubling. I could be wrong about that... if I am, then `ParentObject.visibilities.map {|v| [v,v] }` would work

**Review by orangewolf — DISMISSED** [review]
Looks good on the whole, I think we have an opportunity to reduce some duplication.  I'd also like to see a test of the model validation please. I personally think its fine to skip the view spec for this kind of small change, but others may want to weigh in there.

File: app/models/parent_object.rb (hunk 1 above)
Comment (MaggieZhaoYale) [reply app/models/parent_object.rb]: @orangewolf Thank you, that is a good suggestion. I will add a class method.

Comment (maxkadel) [reply app/models/parent_object.rb]: > Looks good on the whole, I think we have an opportunity to reduce some duplication. I'd also like to see a test of the model validation please. I personally think its fine to skip the view spec for this kind of small change, but others may want to weigh in there.

The general DCE pattern is to use system specs for these types of changes, rather than view specs (I don't have a lot of experience with view specs, but they seem a little implementation-focused to me, which I think is a drawback). I do think adding this to the ParentObject system spec would be valuable in this case.

**Review by maxkadel — CHANGES_REQUESTED** [review]
Agree with Rob's requests, would like the dropdown added to the system spec for ParentObjects.

**Comment by MaggieZhaoYale** [comment]
@maxkadel @orangewolf Thank you. Will commit the change later today.

**Comment by MaggieZhaoYale** [comment]
spec/model/parent_object_spec.rb 208 the order of child_oid is different from the expected. So I slightly modified it.
Screen Shot 2020-09-24 at 1 03 26 PM

File: spec/models/parent_object_spec.rb [hunk 3]
```diff
@@ -235,4 +235,15 @@
       end
     end
   end
+
+  context "with correct visibility value" do
```
Comment (maxkadel) [inline spec/models/parent_object_spec.rb:259]: We need to test for what happens for an empty visibility from the metadata cloud, and for a visibility other than the 3 we've defined in the model is given. Basically, should test the validation method for both nil and random strings.

Also, @orangewolf , should we make the visibilities a private method?

**Review by maxkadel — CHANGES_REQUESTED** [review]

**Review by maxkadel — APPROVED** [review]

**Review by maxkadel — APPROVED** [review]
